### ISSUE TITLE
Sequential quadratic programming does not work with quadcost

### ISSUE THREAD

**Issue by raphaelchinchilla** [jump-dev/Convex.jl#398]
I want to solve a sequence of constrained Quadratic Programming where the matrix H changes. In order to do this, I  use the fix! and free! functions, as explained in the tutorial.  **The issue is that even if I declare H as Semidefinite, Convex.jl does not realize the problem is DCP compliant**

Here is a minimum (not) working example:

```
n=5
b = randn(n) 
x = Variable(n)
H=Semidefinite(n)
Hval=randn(n,n)
Hval.=Hval'*Hval+10*diagm(ones(n)) # symmetric positive definite
fix!(H,Hval)
problem = minimize(x'b +quadform(x,H), [x >= 0])
solve!(problem, SCS.Optimizer(verbose=0))
```

If I try to do this, I get the error `ERROR: MethodError: no method matching quadform(::Variable, ::Variable)`

And if I try to write instead 

`problem = minimize(x'b +x'Hx, [x >= 0])`

I get `Warning: Expression not DCP compliant. Trying to solve non-DCP compliant problems can lead to unexpected behavior.`

and the solution does not work.

**Comment by ericphanson** [comment]
Thanks for the report; it looks like an easy fix. Once #399 passes CI, I'll merge it and tag a release.

**Comment by ericphanson** [comment]
Ok, once https://github.com/JuliaRegistries/General/pull/20132 is merged you should be able to update and use it! Let me know if you run into any other issues.

**Comment by AUK1939** [comment]
Has this been fixed? 

If I run the above code 

    n=5
    b = randn(n) 
    x = Variable(n)
    H=Semidefinite(n)
    Hval=randn(n,n)
    Hval.=Hval'*Hval+10*diagm(ones(n)) # symmetric positive definite
    fix!(H,Hval)
    problem = minimize(x'b +quadform(x,H), [x >= 0])
    solve!(problem, SCS.Optimizer)

The solution is 
    julia> x.value
    5×1 Matrix{Float64}:
     -1.1415298336189401e-5
      0.051775607935912626
     -2.338104123291265e-5
     -1.764813180548554e-5
     -1.8960736530572698e-5

Then I create a new H matrix and fix! it and solve again 

    Hval2=randn(n,n)
    Hval2.=Hval2'*Hval2+10*diagm(ones(n)) # symmetric positive definite
    fix!(H,Hval2)
    solve!(problem, SCS.Optimizer)

But I get the exact same solution (below) as the first problem. Seems like the Hval2 value was ignored. I noticed this on versions 0.15.4 and the master branch (v0.16). Have I missed something or is this a bug? 

     julia> x.value
    5×1 Matrix{Float64}:
    -1.1415298336189401e-5
    0.051775607935912626
    -2.338104123291265e-5
    -1.764813180548554e-5
    -1.8960736530572698e-5

**Comment by ericphanson** [comment]
FYI @raphaelchinchilla, we found out in #585 that my "fix" in #399 of this issue made the error go away but did not actually make this work correctly! I am sorry about that, I hope no results were compromised.

**Comment by raphaelchinchilla** [comment]
Oh man it has been 4 years (and those were pandemic years, so they count double). I have now idea whether results were compromised. Not even sure I remember why I needed that 🤣
